### PR TITLE
Update librarie.min.js & decode search text

### PR DESCRIPTION
**Pull request by alfarok**
### Purpose

[QNTM-4042](https://jira.autodesk.com/browse/QNTM-4042)

Supplemental [Librarie.js PR](https://github.com/DynamoDS/librarie.js/pull/119)

This PR enables all characters and spaces in search and encodes the string passed from librarie.js to Dynamo where the string is decoded and run through search. Previously, spaces were replaced in searches because the results were not accurately returned due to the string formatting. In a later PR all characters that were not a-z 0-9 were removed from the query string to prevent special characters from crashing librarie. This PR should evade those issues.

Looking forward we may want to consider modifying the search timeout logic to improve response fluidity.

### Declarations

Check these if you believe they are true

- [X] The code base is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [X] User facing strings, if any, are extracted into `*.resx` files
- [X] All tests pass using the self-service CI.
- [X] Snapshot of UI changes, if any.
- [X] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@mjkkirschner 

### FYIs

@Racel @jnealb @smangarole

### CODE REVIEW

File: src/LibraryViewExtension/Handlers/SearchResultDataProvider.cs [hunk 1]
```diff
@@ -40,6 +40,8 @@ public override Stream GetResource(IRequest request, out string extension)
             var pathAndQuery = uri.PathAndQuery;
             var index = url.IndexOf(serviceIdentifier);
             var text = url.Substring(index + serviceIdentifier.Length + 1);
+            // unescape query before searching
+            text = Uri.UnescapeDataString(text);
```
Comment (ramramps) [inline src/LibraryViewExtension/Handlers/SearchResultDataProvider.cs]: so, this accepts all special chars?

Comment (alfarok) [reply src/LibraryViewExtension/Handlers/SearchResultDataProvider.cs]: I believe so, here we unescape the encoded URI sent from Librarie.js before searching

**Comment by QilongTang** [comment]
@alfarok Can you talk a bit about where we are after this PR, do we have similar search results with 1.3.3? Or we need to do more work to achieve that

**Comment by alfarok** [comment]
@QilongTang librarie.js seems to be very close, if not 1 to 1 with in-canvas results now.  This doesn't help the lag or performance.  We may want to take a closer look out how the timeout was implemented which creates a delay in displaying the results (much worse in D4R).  I don't think it is very efficient and may cause search to appear slower than it actually is.  @mjkkirschner had some good ideas around improving this logic.

image

image

**Comment by mjkkirschner** [comment]
@alfarok please run the library tests locally I do not know if @smangarole was able to add them to the self serve CI yet.

**Comment by alfarok** [comment]
image

image

**Comment by mjkkirschner** [comment]
LGTM on these.

**Comment by mjkkirschner** [comment]
@alfarok I think creating a version of search that only runs after the search interaction is idle would increase performance a ton - also whenever search gets invoked, we go back and forth between c# and javascript over IPC, then we go back and forth again via an event, then log to analytics ...

**Comment by alfarok** [comment]
image

**Pull request merged**